internal/client/eth_client.go

```diff
@@ -23,31 +30,78 @@ type EthClientInterface interface {
 	BlockByNumber(ctx context.Context, number *big.Int) (*types.Block, error)
 }
 
-// NewEthClient ... Initializer
-func NewEthClient() EthClientInterface {
+func NewEthClient(restyClient *resty.Client) EthClientInterface {
+    if restyClient == nil {
+        restyClient = resty.New()
+    }
 	return &EthClient{
 		client: &ethclient.Client{},
+		restyClient: restyClient,
 	}
 }
```

Comment:
A debug trace client would also be useful here.

Reply:
I added debug logging feature. Please let me know if that's what you had in mind.